### PR TITLE
reimplements headers with better separation of concern

### PR DESCRIPTION
**Pull request by hlminh2000**

### CODE REVIEW

File: modules/components/src/utils/api.js [hunk 1]
```diff
@@ -27,6 +28,7 @@ export const fetchExtendedMapping = ({ graphqlField, projectId }) =>
 
 export const addHeaders = headers => {
   alwaysSendHeaders = { ...alwaysSendHeaders, ...headers };
+  addDownloadHttpHeaders(headers);
```
Comment (alex-wilmer) [inline modules/components/src/utils/api.js]: this is kind of messy in my opinion. Do we need to make these imperative calls? can't we just pass in headers when we actually make calls vs "setting" them and using the set ones later on?

Comment (hlminh2000) [reply modules/components/src/utils/api.js]: Added this issue so we can come back to this later :) : https://github.com/overture-stack/arranger/issues/323

File: modules/components/src/utils/download.js [hunk 2]
```diff
@@ -93,21 +94,31 @@ function createIFrame({ method, url, fields }) {
   return iFrame;
 }
 
-function download({ url, params, method = 'GET' }) {
+function download({ url, params, method = 'GET', customBody = {} }) {
```
Comment (alex-wilmer) [inline modules/components/src/utils/download.js]: can you name this just `body`?

Comment (hlminh2000) [reply modules/components/src/utils/download.js]: sure :)

**Pull request merged**